src/PerfCounters.h

```diff
@@ -126,20 +134,23 @@ class PerfCounters {
   // Only valid while 'counting' is true
   Ticks counting_period;
   pid_t tid;
-  // We use separate fds for counting ticks and for generating interrupts. The
-  // former ignores ticks in aborted transactions, and does not support
-  // sample_period; the latter does not ignore ticks in aborted transactions,
-  // but does support sample_period.
-  ScopedFd fd_ticks_measure;
-  ScopedFd fd_minus_ticks_measure;
-  ScopedFd fd_ticks_interrupt;
-  ScopedFd fd_useless_counter;
-
-  // x86(_64) specific counter to support recording HLE
-  ScopedFd fd_ticks_in_transaction;
-
-  // aarch64 specific counter to detect use of ll/sc instructions
-  ScopedFd fd_strex_counter;
+  struct PerfFds {
+    // We use separate fds for counting ticks and for generating interrupts. The
+    // former ignores ticks in aborted transactions, and does not support
+    // sample_period; the latter does not ignore ticks in aborted transactions,
+    // but does support sample_period.
+    ScopedFd ticks_measure;
+    ScopedFd minus_ticks_measure;
+    ScopedFd ticks_interrupt;
+    ScopedFd useless_counter;
+
+    // x86(_64) specific counter to support recording HLE
+    ScopedFd ticks_in_transaction;
+
+    // aarch64 specific counter to detect use of ll/sc instructions
+    ScopedFd strex_counter;
+  };
+  std::vector<PerfFds> fds;
```

Comment:
Again, please document what this is indexed by.

Reply:
The index is logically the "PMU type", right? But not the kernel's type number, just our own internal numbering of the PMU types.

Reply:
Kind of, I guess I can call it a PMU index and document that each element corresponds to a different type.

It's just that this index isn't stable enough for me to feel comfortable calling it internal numbering..., i.e. the id for the cortex-a55 core PMU type may be different depending on whether it's, alone, or paired with cortex-a75, a76, a77, a78, or a78+x1, and on the order the soc present these to the kernel. All of these being core configurations that actually exists....

Reply:
When you say it's "unstable", you mean across different machines, right? I sure hope it's stable on a given machine.

Reply:
Yes that's right. I didn't try hard to order them but yes they should follow the same order the cores are presented to us.